### PR TITLE
[CPM-71] programme creation api

### PR DESCRIPTION
**Pull request by anian100**
- Programme creation routes for Update, Delete, Get single, Get all, Post
- Changed error messages for course API
- Tests to be done in another ticket

### CODE REVIEW

File: api/src/routes/programme-degree-route.js [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+ *        "204":
+ *           description: No programme degrees found
+ *        "400":
+ *           description: Error message
+ * 
+ */
+router.get("/", async (req, res) => {
+  ProgrammeDegree.find({}, (err, programmeDegrees) => {
+    if (err) {
+      LOGGER.info("Database error thrown");
```
Comment (daltonlim) [inline api/src/routes/programme-degree-route.js:32]: Can you change this to LOGGER.error(err);

File: api/src/routes/programme-degree-route.js [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+ *        "204":
+ *          description: No programme degree for given id
+ *        "400":
+ *          description: Database error
+ *
+ */
+router.get("/:id", async (req, res) => {
+  ProgrammeDegree.findOne({ _id: req.params.id }, (err, result) => {
+    if (err) {
+      LOGGER.info("Database error thrown");
```
Comment (daltonlim) [inline api/src/routes/programme-degree-route.js:77]: as above

**Review by daltonlim — CHANGES_REQUESTED** [review]

File: api/src/routes/programme-degree-route.js (hunk 2 above)
Comment (anian100) [reply api/src/routes/programme-degree-route.js]: done

File: api/src/routes/programme-degree-route.js (hunk 1 above)
Comment (anian100) [reply api/src/routes/programme-degree-route.js]: done

**Review by daltonlim — APPROVED** [review]
LGTM

**Review by wryao64 — APPROVED** [review]
LGTM

**Pull request merged**